### ISSUE TITLE
Using display: none to hide submit button

### ISSUE THREAD

**Issue by samikeijonen** [Automattic/_s#813]
In `style.css` there is CSS like this:

```
/* Search widget. */
.widget_search .search-submit {
	display: none;
}
```

I was wondering if this is accessibility issue or not? In general `display: none` is not a good way to hide text but am I missing something in here.

**Comment by iamdmitrymayorov** [comment]
I believe it is a design decision. Search form will automatically be submitted if you type into it and hit Enter. In this case submit button is not necessary. However, you can easily remove display: none; and style the button the way you like it.

**Comment by corvannoorloos** [comment]
The [`hidden`](https://html.spec.whatwg.org/multipage/interaction.html#the-hidden-attribute) attribute might be suitable here? Although I'm not really sure as I'm generally not using any widgets.

**Comment by obenland** [comment]
@davidakennedy Shouldn't ben issue, should it?

**Comment by davidakennedy** [comment]
This is a good issue to bring up. Generally speaking, most users understand that forms can be submitted when they press the enter key. That's an assumption, of course, but a pretty safe one.

That said, hiding the submit button with `display: none;` isn't as explicit as leaving it visible for everyone, or hiding it offscreen for assistive technology users. For assistive technology users, it's better to have the button there, either visible for everyone, or hidden offscreen in an accessible way.

It's moved offscreen with `screen-reader-text` [in Twenty Fifteen](https://core.trac.wordpress.org/ticket/30110).

If we did anything here, I'd just remove the CSS for the button and leave the decision to theme authors. It borders on a design decision anyway, as @iamdmitrymayorov pointed out.

**Comment by samikeijonen** [comment]
Thanks David for the comment. I was actually looking for Twenty fifteen example but forget to mention it. That's probably how I would do it.

**Comment by iamdmitrymayorov** [comment]
I am for removing `display: none` because now it looks like we are forcing developers to hide the button. Can submit a PR, or perhaps @samikeijonen could do that, as he was first to raise this issue.

**Comment by samikeijonen** [comment]
@iamdmitrymayorov Be my quest for PR if you already have a fork setup. And if this is something others want.

**Comment by obenland** [comment]
Let's go the screen reader text route. Just removing is probably not the best solution.

**Comment by karmatosed** [comment]
Has someone got a PR they can create based on @davidakennedy's feedback?

**Comment by iamdmitrymayorov** [comment]
@karmatosed Are we talking about removing the whole part:
```
/* Search widget. */
.widget_search .search-submit {
    display: none;
}
```
I have a PR ready.

**Comment by iamdmitrymayorov** [comment]
But as mentioned above, there are some benefits in going the .screen-reader-text route. It can serve as an example of hiding elements the accessible way.

**Comment by sixhours** [comment]
Direct link to core patch for Twenty Fifteen, in case it's useful:

https://core.trac.wordpress.org/changeset/30053

**Comment by davidakennedy** [comment]
Related: we show the submit button in Twenty Sixteen:

https://github.com/WordPress/twentysixteen/issues/279
